src/subdomains/user/application/dto/api-key.strategy.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import Strategy from 'passport-headerapikey';
+import { Config } from 'src/config/config';
+
+@Injectable()
+export class HeaderApiKeyStrategy extends PassportStrategy(Strategy, 'api-key') {
```

Comment:
Move this to auth.module (next to jwt.strategy) and rename it to `ApiKeyStrategy`.